SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleaner.java

```diff
@@ -346,7 +345,7 @@ private void drainRequestCleanupQueue() {
           }
           break;
         case DELETING:
-          if (requestWithState.isPresent()) {
+          if (requestWithState.isPresent() && requestWithState.get().getState() != RequestState.DELETING) {
```

Comment:
added this to make sure deleting states aren't ignored and will be cleaned up properly